src/transformers/models/kosmos2/processing_kosmos2.py

```diff
@@ -159,21 +189,18 @@ def __call__(
                     text = f"{self.tokenizer.bos_token}{text}"
                 elif isinstance(text, list):
                     text = [f"{self.tokenizer.bos_token}{s}" for s in text]
-
-            text_encoding = self.tokenizer(
-                text=text,
-                add_special_tokens=(add_special_tokens and add_eos_token),
-                padding=padding and images is None,
-                truncation=truncation,
-                max_length=max_length,
-                pad_to_multiple_of=pad_to_multiple_of if images is None else pad_to_multiple_of,
-                return_attention_mask=return_attention_mask,
-                verbose=verbose,
-                return_tensors=return_tensors if images is None else None,
-                **kwargs,
+            output_kwargs["text_kwargs"]["add_special_tokens"] = (
+                output_kwargs["text_kwargs"]["add_special_tokens"] and add_eos_token
             )
+            output_kwargs["text_kwargs"]["padding"] = padding if images is None else False
```

Comment:
could be 
```suggestion
            output_kwargs["text_kwargs"]["padding"] = padding and (images is None)
```

Reply:
Wouldn't that be a problem when padding is something other than `True` like `"max_length"` ?

Reply:
Should be fine - non-empty strings in boolean expressions evaluate as `True`, but it's just a nit

Reply:
good point, lol. Read too fast - carry on :laughing: